tests/ui/issues/issue-59488.stderr

```diff
@@ -71,6 +71,7 @@ LL |     foo > bar;
    |
    = note: expected fn item `fn() -> i32 {foo}`
               found fn item `fn(i64) -> i64 {bar}`
+   = note: different fn items have unique types, even if their signatures are the same
```

Comment:
We could do the same normalize+"same_type_modulo_infer" check here, only adding this note if the signatures are equal?